File: Algorithm/4/intersection.cpp
#include <iostream>
#include <vector>
#include <set>
#include <unordered_set>

using namespace std;

class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {

        // unordered_set<int> record;
        set<int> record;
        for(int i =0; i < nums1.size(); i++)
            record.insert(nums1[i]);
        // set<int> record(nums1.begin(), nums1.end());

        // unordered_set<int> resultSet;
        set<int> resultSet;
        for(int i = 0; i < nums2.size(); i++)
            if(record.find(nums2[i]) != record.end())
                resultSet.insert(nums2[i]);

        // vector<int> resultVector;
        // for(set<int>::iterator iter = resultSet.begin(); iter != resultSet.end(); iter++)
        //     resultVector.push_back(*iter);
        // return resultVector;
        return vector<int>(resultSet.begin(), resultSet.end());
    }
};

int main(){

    return 0;
}

File: Algorithm/4/intersection.py
def intersection(nums1, nums2):
    """
    :type nums1: List[int]
    :type nums2: List[int]
    :rtype: List[int]
    """
    nums1 = set(nums1)
    nums2 = set(nums2)
    res = []
    if not nums1 or not nums2:
        return []
    for i in nums1:
        if i in nums2:
            res.append(i)
    return res

if __name__ == "__main__":
    nums1 = [1,2,2,1]
    nums2 = [2,2]
    print(intersection(nums1, nums2))
